### PR TITLE
Add test workflow

### PR DESCRIPTION
**Pull request by Sgtpluck**

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+      
+      - name: Install dependencies
```
Comment (Sgtpluck) [inline .github/workflows/tests.yml:41]: should we figure out how to cache these and the precompiled assets?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          POSTGRES_DB: complaint_tracker_test
+          POSTGRES_USER: complaint_tracker 
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_PORT: ${{ job.services.postgres.ports[5432] }}
+        run: |
+          cp config/database.ci.yml config/database.yml
+          bundle exec rake db:create db:migrate
+
+      - name: Precompile assets
+        run: bundle exec rake assets:precompile
```
Comment (rahearn) [inline .github/workflows/tests.yml:57]: I think we can skip this step. It's not needed (normally) for tests, and the circleci ruby buildpack will do it on deploy.

File: .github/workflows/tests.yml (hunk 1 above)
Comment (rahearn) [reply .github/workflows/tests.yml:41]: Yes, but I don't think it needs to be done as part of this PR.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      
+      - name: Run tests
+        env:
+          RAILS_ENV: test
+          POSTGRES_HOST: localhost
+          POSTGRES_DB: complaint_tracker_test
+          POSTGRES_USER: complaint_tracker 
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_PORT: ${{ job.services.postgres.ports[5432] }}
```
Comment (rahearn) [inline .github/workflows/tests.yml:66]: Can we build and pass a `DATABASE_URL` env variable here (and in the create/migrate step too)? `DATABASE_URL` always overrides whatever is in `database.yml`, so we wouldn't need to have a separate ci file that needs to be copied over.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (Sgtpluck) [reply .github/workflows/tests.yml:57]: the request specs seem to require the uswds assets, unfortunately. they're new to me, so i definitely might be setting them up wrong, but if we don't precompile, then the tests choke on 
`@import 'uswds/dist/scss/uswds'`

File: .github/workflows/tests.yml (hunk 3 above)
Comment (Sgtpluck) [reply .github/workflows/tests.yml:66]: sure. is there any documentation you can point me to re: DATABASE_URL? i didn't come across it when i was trying to figure this out!

Comment (rahearn) [reply .github/workflows/tests.yml:66]: https://edgeguides.rubyonrails.org/configuring.html#configuring-a-database section 3.19 and 3.20 go into it.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (rahearn) [reply .github/workflows/tests.yml:57]: I haven't used request specs much before, so disregard my comment!

File: .github/workflows/tests.yml (hunk 3 above)
Comment (Sgtpluck) [reply .github/workflows/tests.yml:66]: thanks!!

**Review by rahearn — APPROVED** [review]
🎉

**Pull request merged**